.github/actions/build-singleapp-if-changed/action.yml

```diff
@@ -0,0 +1,35 @@
+name: Build Single App If Changed
+description: Checks if parent/pom.xml changed and builds single application if needed
+outputs:
+  parent_changed:
+    description: 'Whether parent/pom.xml changed'
```

Comment:
I think this output is not necessary since the info is only used inside this action. As mentioned, can re-use logic from path filter action